benches/public_params.rs

```diff
@@ -37,11 +40,32 @@ fn public_params_benchmark(c: &mut Criterion) {
     });
 }
 
-criterion_group! {
-    name = benches;
-    config = Criterion::default()
-        .measurement_time(Duration::from_secs(120))
-        .sample_size(10);
-    targets = public_params_benchmark
+cfg_if::cfg_if! {
+    if #[cfg(feature = "flamegraph")] {
+        // In order to collect a flamegraph, you need to indicate a profile time, see
+        // https://github.com/tikv/pprof-rs#integrate-with-criterion
+        // Example usage :
+        // cargo criterion --bench public_params --features flamegraph -- --profile-time 5
+        // Warning: it is not recommended to run this on an M1 Mac, as making pprof work well there is hard.
+        criterion_group! {
+            name = benches;
+            config = Criterion::default()
+                .with_profiler(pprof::criterion::PProfProfiler::new(100, pprof::criterion::Output::Flamegraph(None)));
+            targets = public_params_benchmark
+        }
+    } else {
+        criterion_group! {
+            name = benches;
+            config = Criterion::default()
+                .measurement_time(Duration::from_secs(120))
+                .sample_size(10);
+            targets = public_params_benchmark
+        }
+    }
 }
+
+/// To run these benchmarks, first download `criterion` with `cargo install cargo install cargo-criterion`.
+/// Then `cargo criterion --bench public_params`. The results are located in `target/criterion/data/<name-of-benchmark>`.
+/// For flamegraphs, run `cargo criterion --bench public_params --features flamegraph -- --profile-time <secs>`.
+/// The results are located in `target/criterion/profile/<name-of-benchmark>`.
```

Comment:
Ah, dern, this is a macro invocation, to you will probably have to convert this to a simple comment `//`, here and elsewhere.

See https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNUSED_DOC_COMMENTS.html